compiler/luci/service/src/Nodes/CircleReshape.test.cpp

```diff
@@ -37,3 +38,241 @@ TEST(CloneNodeTest, clone_Reshape)
   ASSERT_EQ(node_reshape->newShape()->dim(0), cloned_reshape->newShape()->dim(0));
   ASSERT_EQ(node_reshape->newShape()->dim(1), cloned_reshape->newShape()->dim(1));
 }
+
+TEST(ShapeRuleTest, reshape_both_known_dim)
+{
+  luci::CircleInput input;
+  input.shape({2, 3, 4});
+  input.shape_status(luci::ShapeStatus::VALID);
+
+  luci::CircleConst new_shape;
+  new_shape.dtype(loco::DataType::S32);
+  new_shape.size<loco::DataType::S32>(2);
+  new_shape.at<loco::DataType::S32>(0) = 6;
+  new_shape.at<loco::DataType::S32>(1) = 4;
+  new_shape.shape_status(luci::ShapeStatus::VALID);
+
+  luci::CircleReshape reshape;
+  reshape.tensor(&input);
+  reshape.shape(&new_shape);
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(&reshape, shape));
+
+  ASSERT_EQ(3, input.rank());
+  ASSERT_TRUE(input.dim(0).known());
+  ASSERT_TRUE(input.dim(1).known());
+  ASSERT_TRUE(input.dim(2).known());
+  ASSERT_EQ(2, input.dim(0).value());
+  ASSERT_EQ(3, input.dim(1).value());
+  ASSERT_EQ(4, input.dim(2).value());
```

Comment:
shape inference is to infer output shape from inputs.
I don't think we need to check input shape info.
Input shape check disturbs read the test codes.

Reply:
You're right. Thanks for your feedback.
I'll remove all the assertions not related to output shape inference.